### ISSUE TITLE
Error loading needed lib libzstd.so.1

### ISSUE THREAD

**Issue by sebinho** [ptitSeb/box64#476]
Hi,

I am having issues running a specific 64-bit piece of software. My setup is Ubuntu 22.04.1 LTS (virtual machine running on my MAC M1).
I compiled box64 with the following command:
`cmake .. -DARM_DYNAREC=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo; make -j4`

I get the following errors with my software:

Dynarec for ARM64, with extension: ASIMD AES CRC32 PMULL ATOMICS PageSize:4096 Cores:2
Params database has 11 entries
Box64 with Dynarec v0.2.1 73467dc5 built on Dec 15 2022 11:56:55
Using default BOX64_LD_LIBRARY_PATH: ./:lib/:lib64/:x86_64/:bin64/:libs64/
Using default BOX64_PATH: ./:bin/
Counted 76 Env var
...
...
...
Error loading needed lib libzstd.so.1
Error loading one of needed lib
Error initializing needed lib libefx.so
Error loading one of needed lib
Error initializing needed lib libarchfpga.so
Error loading one of needed lib
Error initializing needed lib libvpr.so
Error loading one of needed lib
Error: loading needed libs in elf /home/neo/eda/efinity/2022.1/bin/efinity

Is here any way to make box64 work for libzstd? 
Thanks a lot for your help!

**Comment by sebinho** [comment]
I managed to get it working by copying 'libzstd.so.1' from another computer I have and putting in on my ARM linux.
Anyways it would be awesome to have this library wrapped by default inside box64.
Thx!

**Comment by ptitSeb** [comment]
I'll check. there are a few complicated function to wrap IIRC, so it might not be super trivial to do. It's rarely used, so I would need a software that use it to check the wrapping is correct.

**Comment by sebinho** [comment]
Indeed. Here I am using a very specific software that not many people use and it is not free. Not sure which other software is using it.

**Comment by michele-perrone** [comment]
> I managed to get it working by copying 'libzstd.so.1' from another computer I have and putting in on my ARM linux. Anyways it would be awesome to have this library wrapped by default inside box64. Thx!

Hi, I have the same problem with another library.

Where did you copy the library exactly in your ARM Linux installation?

**Comment by LukeShortCloud** [comment]
@sebinho The zstandard library is now partially wrapped.

https://github.com/ptitSeb/box64/blob/main/src/wrapped/wrappedzstd_private.h

@michele-perrone You can copy x86_64 libraries to `/usr/lib/box64-x86_64-linux-gnu/` and x86 libraries to `/usr/lib/box64-i386-linux-gnu/`. Alternatively, you can use the `BOX64_LD_LIBRARY_PATH` environment variable to specify custom paths.

Closing as resolved.
